client/src/components/Kanban/CardForm/CardForm.tsx

```diff
@@ -0,0 +1,136 @@
+import { Box, Button, Divider, InputBase, Grid, Typography, useTheme } from '@material-ui/core';
+import { Dispatch } from 'react';
+import { useState, useEffect, SetStateAction } from 'react';
+import { useKanban } from '../../../context/useKanbanContext';
+import useColorTagStyles from '../shared/colorStyles';
+import useStyles from './useStyles';
+
+type CardFormProps = {
+  columnId: string;
+};
+
+const CardForm = ({ columnId }: CardFormProps): JSX.Element => {
+  const classes = useStyles();
+  const [formIsOpen, setFormIsOpen] = useState<boolean>(false);
+
+  const openForm = (): void => setFormIsOpen(true);
+  const closeForm = (): void => setFormIsOpen(false);
+
+  const handleClose = (event: globalThis.KeyboardEvent) => {
+    if (event.key === 'Escape') {
+      closeForm();
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('keydown', handleClose, false);
+    return () => {
+      document.removeEventListener('keydown', handleClose, false);
+    };
+  }, []);
+
+  return (
+    <Box>
+      {formIsOpen ? <InnerForm columnId={columnId} formAction={setFormIsOpen} /> : null}
+      {!formIsOpen && (
+        <Button
+          onClick={openForm}
+          size="large"
+          color="secondary"
+          disableElevation
+          className={`${!formIsOpen ? classes.button : ''}`}
+        >
+          Add a card...
+        </Button>
+      )}
+    </Box>
+  );
+};
+
+type InnerFormProps = {
+  columnId: string;
+  formAction: Dispatch<SetStateAction<boolean>>;
+};
+const InnerForm = ({ columnId, formAction }: InnerFormProps) => {
+  const [name, setName] = useState<string>('');
+  const [selectedTagColor, setTagColor] = useState<string>('white');
+  const { addCard } = useKanban();
+  const theme = useTheme();
+  const classes = useStyles();
+  return (
+    <>
+      <Box className={classes.cardFormWrapper}>
+        <Box className={classes.cardForm}>
+          <InputBase
+            className={classes.input}
+            value={name}
+            onChange={(e) => setName(e.target.value)}
+            placeholder="Add title..."
+          />
+        </Box>
+        <Divider light />
+        <Box className={classes.tagWrapper}>
+          <Grid container alignItems="center" justify="space-between">
+            <Grid item>
+              <Typography className={classes.typography} variant="body1">
+                Select tag:
+              </Typography>
+            </Grid>
+            <Grid item>
+              <Grid container className={classes.colorsWrapper}>
+                {Object.keys(theme.palette.tags).map((tagColor: string): JSX.Element => {
+                  return (
+                    <Color
+                      key={tagColor}
```

Comment:
Can multiple forms be open? If so, this might not be unique enough of a key, you might want to do something like:
```javascript
key=`${columnId}-${tagColor}`
```

Reply:
That's a good point!